### PR TITLE
Config fixes

### PR DESCRIPTION
**Pull request by psinger**
Closes https://github.com/h2oai/h2o-llmstudio/issues/60

Also removes api keys from cfg and moves them to pure env variables

@maxjeblick please make sure the visibility changes I made are OK - they are needed to fix `pretrained_weights` stuff

### CODE REVIEW

File: app_utils/utils.py [hunk 1]
```diff
@@ -1571,9 +1571,10 @@ def start_experiment(cfg: Any, q: Q, pre: str, gpu_list: Optional[List] = None)
     process_queue = list(set(all_process_queue))
 
     secrets = {
-        "NEPTUNE_API_TOKEN": cfg.logging.neptune_api_token,
-        "OPENAI_API_KEY": cfg.environment.openai_api_token,
+        "NEPTUNE_API_TOKEN": q.client["default_neptune_api_token"],
+        "OPENAI_API_KEY": q.client["default_openai_api_token"]
     }
+    print(secrets)
```
Comment (maxjeblick) [inline app_utils/utils.py:1577]: Artifact from debugging?

File: llm_studio/src/utils/utils.py [hunk 2]
```diff
@@ -31,16 +31,11 @@ def set_seed(seed: int = 1234) -> None:
 
 def set_environment(cfg):
     """Sets and checks environment settings"""
-    if not cfg.environment.openai_api_token:
-        cfg.environment.openai_api_token = os.getenv("OPENAI_API_KEY", "")
-    if not cfg.logging.neptune_api_token:
-        cfg.logging.neptune_api_token = os.getenv("NEPTUNE_API_TOKEN", "")
-    os.environ["OPENAI_API_KEY"] = cfg.environment.openai_api_token
-    openai.api_key = cfg.environment.openai_api_token
```
Comment (maxjeblick) [inline llm_studio/src/utils/utils.py:39]: This line is no longer needed?

File: llm_studio/src/utils/config_utils.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def convert_cfg_to_nested_dictionary(cfg: ConfigProblemBase) -> dict:
 
     for k, v in cfg_dict.items():
 
-        if k.startswith("_") or cfg._get_visibility(k) < 0:
```
Comment (maxjeblick) [inline llm_studio/src/utils/config_utils.py:109]: What's the rationale behind removing `cfg._get_visibility(k) < 0`?

With this change, yaml config will also save non visible configuration items that are of type `KNOWN_TYPE_ANNOTATIONS`. 
I don't think it's necessarily a disadvantage to also store those fields, as they are part of the configuration.

Comment (psinger) [reply llm_studio/src/utils/config_utils.py:109]: It is needed becasue we set `pretrained_weights` when activating `Use previous experiment weights` but it gets visibility -1 so the yaml does not store it. We can either add it explicitly here as another exception, but I thought it should not hurt to also store those not being visible.
wdyt?

File: llm_studio/src/utils/utils.py (hunk 2 above)
Comment (psinger) [reply llm_studio/src/utils/utils.py:39]: We are setting it here, or?
`openai.api_key = os.getenv("OPENAI_API_KEY", "")`

Comment (maxjeblick) [reply llm_studio/src/utils/utils.py:39]: Ah yes missed the lower part

File: llm_studio/src/utils/config_utils.py (hunk 3 above)
Comment (maxjeblick) [reply llm_studio/src/utils/config_utils.py:109]: Yes, I think it's good to store all configurations, as visibility options may change.

**Review by maxjeblick — APPROVED** [review]
LGTM thanks!

**Pull request merged**